src/array.jl

```diff
@@ -730,26 +823,28 @@ function Base.reshape(A::CategoricalArray{T, N}, dims::Dims) where {T, N}
 end
 
 """
-    categorical{T}(A::AbstractArray{T}[, compress::Bool]; ordered::Bool=false)
+    categorical{T}(A::AbstractArray{T}[, compress::Bool]; levels=nothing, ordered=false)
 
 Construct a categorical array with the values from `A`.
 
-If the element type supports it, levels are sorted in ascending order;
-else, they are kept in their order of appearance in `A`. The `ordered` keyword
-argument determines whether the array values can be compared according to the
-ordering of levels or not (see [`isordered`](@ref)).
+The `levels` keyword argument can be a vector specifying possible values for the data
+(this is equivalent to but more efficient than calling [`levels!`](@ref)
+on the resulting array).
+If `levels` is omitted and the element type supports it, levels are sorted
+in ascending order; else, they are kept in their order of appearance in `A`.
+The `ordered` keyword argument determines whether the array values can be compared
+according to the ordering of levels or not (see [`isordered`](@ref)).
 
 If `compress` is provided and set to `true`, the smallest reference type able to hold the
 number of unique values in `A` will be used. While this will reduce memory use, passing
 this parameter will also introduce a type instability which can affect performance inside
 the function where the call is made. Therefore, use this option with caution (the
 one-argument version does not suffer from this problem).
 
-    categorical{T}(A::CategoricalArray{T}[, compress::Bool]; ordered::Bool=isordered(A))
+    categorical(A::CategoricalArray, compress::Bool]; levels=nothing, ordered=false)
```

Comment:
I understand essentially this is the API (even if internally it is implemented differently)
```suggestion
    categorical(A::CategoricalArray, compress::Bool=false; levels=nothing, ordered=false)
```

Reply:
I think it is OK to leave it as is. I just wanted to be sure that I understand the things correctly.